RFCs/FS-1090-Generic-struct-type-whose-fields-are-all-unmanaged-types-is-unmanaged.md

```diff
@@ -0,0 +1,198 @@
+# F# RFC FS-1088 - (Generic struct type whose fields are all unmanaged types is unmanaged)
+
+The design suggestion [Generic struct type whose fields are all unmanaged types is unmanaged](https://github.com/fsharp/fslang-suggestions/issues/692) has been marked "approved in principle".
+This RFC covers the detailed proposal for this suggestion.
+
+* [x] Approved in principle
+* [x] [Suggestion](https://github.com/fsharp/fslang-suggestions/issues/692)
+* [ ] Implementation: [In progress](https://github.com/dotnet/fsharp/pull/6064)
+
+# Summary
+
+[summary]: #summary
+
+Allow generic structs to be `unmanaged` at construction if all fields are `unmanaged` types.
+
+# Motivation
+
+[motivation]: #motivation
+
+* Make it easier to author low level interop code.
+* Improve interoperability with C#.
+
+# Detailed design
+
+[design]: #detailed-design
+
+Currently, it is not possible to have an `unmanaged` constraint working on generic struct types.
```

Comment:
This will need additional clarification:

```suggestion
Currently, it is not possible to define a generic struct with the `unmanaged` constraint and substitute the type with another generic struct, even if its type is also constrained to be `unmanaged`.
```